tests/amazon/test_credentials.py

```diff
@@ -31,10 +31,11 @@
 
         aws_creds = credentials.session.get_credentials()
 
-        # Moto provides the following credentials from 169.254.269.254
-        self.assertEquals(aws_creds.access_key, 'test-key')
-        self.assertEquals(aws_creds.secret_key, 'test-secret-key')
-        self.assertEquals(aws_creds.token, 'test-session-token')
+        # Moto provides 'test-*' via instance_metadata mock
+        # Moto provides '*EXAMPLE*' via sts mock
+        self.assertTrue(aws_creds.access_key, ['test-key', 'AKIAIOSFODNN7EXAMPLE'])
+        self.assertTrue(aws_creds.secret_key in ['test-secret-key', 'aJalrXUtnFEMI/K7MDENG/bPxRfiCYzEXAMPLEKEY'])
+        self.assertTrue(aws_creds.token in ['test-session-token', 'BQoEXAMPLEH4aoAH0gNCAPyJxz4BlCFFxWNE1OPTgk5TthT+FvwqnKwRcOIfrRh3c/LTo6UDdyJwOOvEVPvLXCrrrUtdnniCEXAMPLE/IvU1dYUg2RVAJBanLiHb4IgRmpRV3zrkuWJOgQs8IZZaIv2BXIa2R4OlgkBN9bkUDNCJiBeb/AXlzBBko7b15fjrBs2+cTQtpZ3CYWFXG8C5zqx37wnOE49mRl/+OtkIKGO7fAE'])
```

Comment:
Yep, see that. There were some vars missing from the os.environ purge which were/could cause issues.